mlflow/genai/experimental/databricks_trace_archival.py

```diff
@@ -281,7 +313,12 @@ def _do_enable_databricks_archival(
     Raises:
         MlflowException: If any step of the archival process fails
     """
-    trace_archival_location = f"{catalog}.{schema}.{table_prefix}_{experiment_id}"
+    from databricks.sdk import WorkspaceClient
+
+    workspace_id = WorkspaceClient().get_workspace_id()
```

Comment:
The same code is used in `_resolve_ingest_url`, let's unify them as a single helper function